src/timer.rs

```diff
@@ -0,0 +1,384 @@
+
+                /// Clear interrupt for match register 0.
+                pub fn clear_match0_interrupt(&self) {
+                    let timer = unsafe { &*pac::TIMER::ptr() };
+                    timer.[<ticr $channel>].modify(|_r, w| w.tclr_0().set_bit());
+                }
+
+                /// Clear interrupt for match register 1.
+                pub fn clear_match1_interrupt(&self) {
+                    let timer = unsafe { &*pac::TIMER::ptr() };
+                    timer.[<ticr $channel>].modify(|_r, w| w.tclr_1().set_bit());
+                }
+
+                /// Clear interrupt for match register 2.
+                pub fn clear_match2_interrupt(&self) {
+                    let timer = unsafe { &*pac::TIMER::ptr() };
+                    timer.[<ticr $channel>].modify(|_r, w| w.tclr_2().set_bit());
+                }
+
+                /// Sets when the to preload.
+                pub fn set_preload(&self, preload: Preload) {
+                    let timer = unsafe { &*pac::TIMER::ptr() };
+                    timer
+                        .[<tplcr $channel>]
+                        .modify(|_r, w| unsafe { w.tplcr().bits(preload.to_prlcr()) });
+                }
+
+                /// Sets match register 0
+                pub fn set_match0(&self, time: Milliseconds) {
+                    let time = (self.clock.0 as u128 * time.0 as u128 / 1_000u128) as u32;
+                    let timer = unsafe { &*pac::TIMER::ptr() };
+                    timer.[<tmr $channel _0>].modify(|_r, w| unsafe { w.tmr().bits(time) });
+                }
+
+                /// Sets match register 1
+                pub fn set_match1(&self, time: Milliseconds) {
+                    let time = (self.clock.0 as u128 * time.0 as u128 / 1_000u128) as u32;
+                    let timer = unsafe { &*pac::TIMER::ptr() };
+                    timer.[<tmr $channel _1>].modify(|_r, w| unsafe { w.tmr().bits(time) });
+                }
+
+                /// Sets match register 2
+                pub fn set_match2(&self, time: Milliseconds) {
+                    let time = (self.clock.0 as u128 * time.0 as u128 / 1_000u128) as u32;
+                    let timer = unsafe { &*pac::TIMER::ptr() };
+                    timer.[<tmr $channel _2>].modify(|_r, w| unsafe { w.tmr().bits(time) });
+                }
+
+                // Current counter value in raw ticks.
+                pub fn current_ticks(&self) -> u32 {
+                    let timer = unsafe { &*pac::TIMER::ptr() };
+                    timer.[<tcr $channel>].read().bits()
+                }
+
+                // Current counter value in milliseconds.
+                pub fn current_time(&self) -> Milliseconds {
+                    let timer = unsafe { &*pac::TIMER::ptr() };
+                    let ticks = timer.[<tcr $channel>].read().bits() as u128;
+                    Milliseconds::new(((ticks * 1_000u128 / self.clock.0 as u128 * 1_000u128) / 1_000u128) as u32)
```

Comment:
Not sure I'm following the arithmetic here.
Assuming the clock source is 32khz, if the read value was 3200 I'd expect a return value here of 100ms.
To state that mathematically:
`ms = (ticks / self.clock.0) * 1000`
though to avoid range/precision losses you could promote then multiply then divide
`ticks as u64 * 1000u64 / self.clock.0 as u64`